packages/superset-ui-chart/src/components/SuperChart.tsx

```diff
@@ -92,27 +141,34 @@ export default class SuperChart extends React.PureComponent<Props, {}> {
   }
 
   render() {
-    const { width: inputWidth, height: inputHeight } = this.props as PropsWithDefault;
-
-    // Parse them in case they are % or 'auto'
-    const widthInfo = parseLength(inputWidth);
-    const heightInfo = parseLength(inputHeight);
+    const { heightInfo, widthInfo, BoundingBox } = this.parseDimension(this
+      .props as PropsWithDefault);
 
     // If any of the dimension is dynamic, get parent's dimension
     if (widthInfo.isDynamic || heightInfo.isDynamic) {
       const { debounceTime } = this.props;
 
       return (
-        <ParentSize debounceTime={debounceTime}>
-          {({ width, height }) =>
-            width > 0 &&
-            height > 0 &&
-            this.renderChart(
-              widthInfo.isDynamic ? Math.floor(width * widthInfo.multiplier) : widthInfo.value,
-              heightInfo.isDynamic ? Math.floor(height * heightInfo.multiplier) : heightInfo.value,
-            )
-          }
-        </ParentSize>
+        // bounding box will ensure that when
```

Comment:
could probably consolidate to fewer lines here? I also wonder if this could be moved to the `BoundingBox` selector to give more context to the `Fragment` vs `div` logic?

Reply:
Comments moved as suggested.